src/core_landice/mode_forward/mpas_li_iceshelf_melt.F

```diff
@@ -1128,6 +1140,345 @@ subroutine iceshelf_melt_ismip6(domain, err)
+         call mpas_pool_get_array(geometryPool, 'meltingThickness', meltingThickness)
+         call mpas_pool_get_array(geometryPool, 'bedTopography', bedTopography)
+         call mpas_pool_get_array(geometryPool, 'lowerSurface', lowerSurface) 
+         call mpas_pool_get_array(geometryPool, 'groundedMarineMarginMask', groundedMarineMarginMask)
+         call mpas_pool_get_array(geometryPool, 'faceMeltRateApplied', faceMeltRateApplied)
+         call mpas_pool_get_array(geometryPool, 'faceMeltSpeed', faceMeltSpeed)
+         
+       ! Calculate retreat rate for each cell
+         do iCell = 1, nCellsSolve
+            ! Define marine marginal cells as those that are (1) at the ice
+            ! margin, (2) have at least one neighboring cell without ice, (3) contain
+            ! grounded ice, and (4) have bed topography below sea level.
+            ! OR is adjacent to an inactive floating margin cell
+            nEmptyNeighbors = 0
+            oceanEdgeLength = 0.0_RKIND
+            ! Check if neighboring cells contain ice and have bed topo below sea
+            ! level
+            do iEdge = 1, nEdgesOnCell(iCell)
+               iNeighbor = cellsOnCell(iEdge, iCell)
+                 if ( ((thickness(iNeighbor) == 0.0_RKIND) & 
+                    .and. iNeighbor > 0 & ! make sure this cell exists
+                    .and. iNeighbor <= nCellsSolve & ! make sure this cell exists
+                    .and. bedTopography(iNeighbor) < seaLevel) &
+                    .or. (li_mask_is_floating_ice(cellMask(iNeighbor)) & 
+                    .and. li_mask_is_margin(cellMask(iNeighbor)) &
+                    .and. (.not. li_mask_is_albany_active(cellMask(iNeighbor)))) ) then
+                     nEmptyNeighbors = nEmptyNeighbors + 1
+                     oceanEdgeLength = oceanEdgeLength + dvEdge(edgesOnCell(iEdge, iCell))
+                 endif
+            enddo 
+            
+            if ( nEmptyNeighbors > 0 &
+               .and. thickness(iCell) > 0.0_RKIND &
+               .and. li_mask_is_grounded_ice(cellMask(iCell)) &
+               .and. bedTopography(iCell) < seaLevel ) then
+                
+                groundedMarineMarginMask(iCell) = 1
+                waterDepth = seaLevel - bedTopography(iCell)
+                          
+                submergedFaceArea = waterDepth * oceanEdgeLength
+                
+                if (config_uniform_face_melt_rate == 0.0) then                
+                ! Calculate ice front melt rate at each cell
+                   faceMeltSpeed(iCell) = (aSubglacial * waterDepth * & ! m s^-1
+                         (ismip6Runoff(iCell) / rhow * secPerDay)**alphaSubglacial &
+                         + B) * (TFocean(iCell) + addTFocean)**betaTF / secPerDay
+                else
+                   faceMeltSpeed(iCell) = config_uniform_face_melt_rate
+                endif
+                !calvingVelocity(iCell) = calvingVelocity(iCell) + iceFrontMeltRate ! m s^-1
```

Comment:
Do you feel comfortable removing this now?  Or want to keep it in awhile longer until things are fully settled?